### ISSUE TITLE
jansson doesn't compile with MinGW

### ISSUE THREAD

**Issue by qno** [bincrafters/community#774]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **jansson/2.11@bincrafters/stable**
  * Operating System+version: **Windows Server 2016**
  * Compiler+version: **gcc-8.1**
  * Conan version: **conan 1.16**
  * Python version: **Python 3.72**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=8.1
compiler.exception=seh
compiler.libcxx=libstdc++11
compiler.threads=posix
build_type=Release
[options]
[build_requires]
mingw_installer/1.0@conan/stable
msys2_installer/latest@bincrafters/stable
[env]
```

### Steps to reproduce (Include if Applicable)
Build jansson recipe with MinGW 

### Logs (Include/Attach if Applicable)
[log_27_445.zip](https://github.com/bincrafters/community/files/3154798/log_27_445.zip)

https://github.com/bincrafters/conan-jansson/blob/stable/2.11/conanfile.py#L48 - it seems that building the tests just have to be disabled, as the libjanson was already compiled before the error.

```
...
[ 23%] Building C object source_subfolder/CMakeFiles/jansson.dir/src/utf.c.obj
[ 26%] Building C object source_subfolder/CMakeFiles/jansson.dir/src/value.c.obj
[ 28%] Linking C static library ..\lib\libjansson.a
[ 28%] Built target jansson
Scanning dependencies of target test_array
...
```
So i would assume, that 
`cmake.definitions["JANSSON_WITHOUT_TESTS"] = False`
must be changed to
`cmake.definitions["JANSSON_WITHOUT_TESTS"] = True`

Or just build in MinGW  with autotools.

**Comment by qno** [comment]
I actually checked this on my local machine with adapted recipe and with disabled test building the package succeeded.

**Comment by Croydon** [comment]
I have just published version 2.12 of jansson. Are you stuck with 2.11 or could you update? If you are stuck I could backport the build additions

AppVeyor doesn't provide MinGW 9, but I added 6, 7, 8 builds to it

**Comment by Croydon** [comment]
I have just published version 2.12 of jansson. Are you stuck with 2.11 or could you update? If you are stuck I could backport the build additions

AppVeyor doesn't provide MinGW 9, but I added 6, 7, 8 builds to it

**Comment by qno** [comment]
@Croydon - and do you mean by adding MinGW only for jannson build? So in my project I use a lot more conan packages and for building the project it has to compile any package for MinGW. So it should turned on in general for building.

**Comment by Croydon** [comment]
I have just published version 2.12 of jansson. Are you stuck with 2.11 or could you update? If you are stuck I could backport the build additions

AppVeyor doesn't provide MinGW 9, but I added 6, 7, 8 builds to it

**Comment by Croydon** [comment]
I have just published version 2.12 of jansson. Are you stuck with 2.11 or could you update? If you are stuck I could backport the build additions

AppVeyor doesn't provide MinGW 9, but I added 6, 7, 8 builds to it

**Comment by Croydon** [comment]
For some reason GitHub has some weird behaviour right now. Comments seem to get posted after a very long delay and I don't see them chronological anymore.....

Anyway...

______

> @Croydon - and do you mean by adding MinGW only for jannson build? So in my project I use a lot more conan packages and for building the project it has to compile any package for MinGW. So it should turned on in general for building.

Yes. We can add e.g. MinGW builds or builds for very old compilers, or the ARM architecture on demand, but we simply don't have the resources to build all recipes for all possible configurations.

**Comment by Croydon** [comment]
The issue is fixed.
MinGW builds are available for 2.12

**Comment by qno** [comment]
> I have just published version 2.12 of jansson. Are you stuck with 2.11 or could you update? If you are stuck I could backport the build additions
> 
> AppVeyor doesn't provide MinGW 9, but I added 6, 7, 8 builds to it

Thank you. I switched to 2.12 and it builds now for me (still no MinGW artifacts available).
